api/v1/issues/IssuesHandler.inc.php

```diff
@@ -0,0 +1,175 @@
+<?php 
+
+/**
+ * @file api/v1/issues/IssuesHandler.inc.php
+ *
+ * Copyright (c) 2014-2016 Simon Fraser University Library
+ * Copyright (c) 2003-2016 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class IssuesHandler
+ * @ingroup api_v1_issues
+ *
+ * @brief Handle API requests for issues operations.
+ *
+ */
+
+import('lib.pkp.classes.handler.APIHandler');
+import('classes.core.ServicesContainer');
+
+class IssuesHandler extends APIHandler {
+	
+	/**
+	 * Constructor
+	 */
+	public function __construct() {
+		$this->_handlerPath = 'issues';
+		$roles = array(ROLE_ID_MANAGER, ROLE_ID_SUB_EDITOR, ROLE_ID_ASSISTANT, ROLE_ID_REVIEWER, ROLE_ID_AUTHOR);
+		$this->_endpoints = array(
+			'GET' => array (
+				array(
+					'pattern' => $this->getEndpointPattern(),
+					'handler' => array($this,'getIssueList'),
+					'roles' => $roles
+				),
+				array(
+					'pattern' => $this->getEndpointPattern().  '/{issueId}',
+					'handler' => array($this,'getIssue'),
+					'roles' => $roles
+				),
+			)
+		);
+		parent::__construct();
+	}
+	
+	//
+	// Implement methods from PKPHandler
+	//
+	function authorize($request, &$args, $roleAssignments) {
+		$routeName = null;
+		$slimRequest = $this->getSlimRequest();
+	
+		if (!is_null($slimRequest) && ($route = $slimRequest->getAttribute('route'))) {
+			$routeName = $route->getName();
+		}
+		
+		import('lib.pkp.classes.security.authorization.ContextRequiredPolicy');
+		$this->addPolicy(new ContextRequiredPolicy($request));
+		
+		import('classes.security.authorization.OjsJournalMustPublishPolicy');
+		$this->addPolicy(new OjsJournalMustPublishPolicy($request));
+		
+		if ($routeName === 'getIssue') {
+			import('classes.security.authorization.OjsIssueRequiredPolicy');
+			$this->addPolicy(new OjsIssueRequiredPolicy($request, $args));
+		}
+		
+		return parent::authorize($request, $args, $roleAssignments);
+	}
+	
+	//
+	// Public handler methods
+	//
+	/**
+	 * Handle file download
+	 * @param $slimRequest Request Slim request object
+	 * @param $response Response object
+	 * @param array $args arguments
+	 * @return Response
+	 */
+	public function getIssueList($slimRequest, $response, $args) {
+		$request = $this->getRequest();
+		$context = $request->getContext();
+		$journal = $request->getJournal();
+		$data = array();
+		
+		$volume = $this->getParameter('volume', null);
+		$number = $this->getParameter('number', null);
+		$year = $this->getParameter('year', null);
+		
+		$issueDao = DAORegistry::getDAO('IssueDAO');
+		$issues = $issueDao->getPublishedIssuesByNumber($journal->getId(), $volume, $number, $year);
+		
+		while ($issue = $issues->next()) {
+			$data[] = array(
+				'id'				=> $issue->getBestIssueId(),
+				'title'				=> $issue->getLocalizedTitle(),
+				'series'			=> $issue->getIssueSeries(),
+				'datePublished'		=> $issue->getDatePublished(),
+				'lastModified'		=> $issue->getLastModified(),
```

Comment:
Make sure your editor is configured to use tabs for indentation, and display them as 8 columns each.